### ISSUE TITLE
Cannot use freeze_graph.py and optimize_for_inference.py with any model 

### ISSUE THREAD

**Issue by Swordancer** [tensorflow/tensorflow#18877]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10 x64
- **TensorFlow installed from (source or binary):**: PIP Python 3.6
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**: 3.6
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: .9.0
- **GPU model and memory**: 2GB
- **Exact command to reproduce**:

> py -m tensorflow.python.tools.freeze_graph --input_graph=output_graph.pb --input_checkpoint=C:/tmp/_retrain_checkpoint --output_graph=/tmp/frozen_graph.pb --output_node_names=final_result --input_binary=true

> py "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python36_64\Lib\site-packages\tensorflow\python\tools\optimize_for_inference.py" --input=output_graph.pb --output=/tmp/optimazed_frozen_graph.pb --frozen_graph=true --input_names="Placeholder" --output_names "final_result" --inference_type=float

### Describe the problem

I have generated model with this tutorial for my own classes: https://www.tensorflow.org/tutorials/image_retraining

I cannot use generated .pb file with EmguCV or OpenCV, becouse of import error, it looks like some layers are unsupported,  so I was trying to run these scripts first:

freeze_graph.py

> py C:\tensorflow-master\tensorflow\python\tools\freeze_graph.py --input_graph=output_graph.pb --input_checkpoint=C:/tmp/_retrain_checkpoint --output_graph=/tmp/frozen_graph.pb --output_node_names=softmax --input_binary=true

I have an error:

Names_to_saveables must be a dict mapping string names to Tensors/Variables. Not a variable: Tensor("final_retrain_ops/biases/final_biases:0", shape=(6,), dtype=float32)

Then optimize_for_inference.py 

Multiple errors 

> WARNING:tensorflow:Didn't find expected Conv2D input to 'module_apply_default/MobilenetV2/expanded_conv_16/depthwise/BatchNorm/FusedBatchNorm'

Scripts doesn't work with any generated .pb file, even some I've downloaded from the Internet

It looks like im doing something wrong or these scripts doesn't work with model from tutorial. I there any solution for that?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from

**Comment by LucasMahieu** [comment]
Thinks it is related to this https://github.com/tensorflow/tensorflow/issues/16545#issuecomment-382204169

**Comment by skye** [comment]
@petewarden are you the right person to address this?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by JooYongLee** [comment]
I have a same problem with it. Without Batch-Norm Layer, I can import created *.pb model. But with batch-norm layer, I cannot import *pb file using opencv function "ReadNetFromTensorflow". If you solve this problem, please tell me how to fix it.

**Comment by Regilan** [comment]
optimize_for_inference \ --input=frozen_mnist.pb \ --output=opt_mnist_graph.pb \ --frozen_graph=True \ --input_names=input_tensor \ --output_names=softmax_tensor
https://github.com/LucasMahieu The above thing is my output when i command to optimize the freezed_graph! I got freezing the model.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 151 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @petewarden: It has been 166 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by petewarden** [comment]
I believe this error usually happens when a graph that's already been frozen is fed into freeze_graph.py, and it's been a while since any updates, so closing this issue.
